### ISSUE TITLE
Add license validation and handling

### ISSUE THREAD

**Issue by znichollscr** [climate-resource/input4mips_validation#19]
## The motivation

As a data producer, I want licence information to be filled out for me rather than having to type lots of stuff.

## The proposed solution

Use the CVs to handle licence stuff.

## Additional context

It looks like the CMOR tables are going to simplify this considerably, hence this is blocked until they are settled.

https://github.com/PCMDI/mip-cmor-tables/blob/jsonld/JSONLD/license/cc-by-sa-4.0.json

**Comment by durack1** [comment]
Yep, so the default license is CC-BY-4.0, with a couple of variants also allowed - the variants we had included in CMIP6 are included at https://github.com/WCRP-CMIP/CMIP6_CVs/blob/main/CMIP6_license.json, with a strong preference to avoid the NC and SA variants, as this complicates downstream data use
